install.packages()
installed.packages()
source("http://bioconductor.org/biocLite.R")
biocLite("edgeR")
library(limma)
getwd()
setwd("C:/Users/JiyingLi/Desktop/project221")
load("C:/Users/JiyingLi/Desktop/project221/EdgeR.RData")
y$samples
y$samples$group
levels(y$samples$group)
getwd()
setwd("C:/Users/JiyingLi/Desktop/project221")
spacefly.counts<- read.csv("flies_space_counts.csv")
head(spacefly.counts)
rownames(spacefly.counts)<-spacefly.counts$Name
colnames(spacefly.counts)<-c("G3R3","G1R2","G1R3","G1R4","G3R1","G3R2","G1R1","G3R4")
head(spacefly.counts)
View(spacefly.counts)
spacefly.counts.table<-spacefly.counts[,c(2:9)]
class(spacefly.counts.table)
group <- factor(c("G3R3","G1R2","G1R3","G1R4","G3R1","G3R2","G1R1","G3R4"))
design <- model.matrix(~group)
design
y<-DGEList(counts=spacefly.counts.table,group=group) #synopsis: DGEList(counts= x, group = y)
y$samples  # sample should be a variable in DEGList result.
levels(y$samples$group) # so was group
bcv <- 0.4
CPM <- cpm(y)
CPM_table<-cbind(CPM,rownames(CPM))
CPM_table<-as.data.frame(CPM_table)
plotMDS(y)
View(spacefly.counts.table)
View(CPM)
library(edgeR)
library(limma)
setwd("C:/Users/JiyingLi/Desktop/project221")
spacefly.counts<- read.csv("flies_space_counts.csv")
View(spacefly.counts)
rownames(spacefly.counts)<-spacefly.counts$Name
colnames(spacefly.counts)<-c("Gene"，"G3R3","G1R2","G1R3","G1R4","G3R1","G3R2","G1R1","G3R4")
View(spacefly.counts)
spacefly.counts.table<-spacefly.counts[,c(2:9)]
View(spacefly.counts.table)
class(spacefly.counts.table)
group <- factor(c("G3R3","G1R2","G1R3","G1R4","G3R1","G3R2","G1R1","G3R4"))
design <- model.matrix(~group)
y<-DGEList(counts=spacefly.counts.table,group=group)
bcv <- 0.4
CPM <- cpm(y)
CPM_table<-cbind(CPM,rownames(CPM))
CPM_table<-as.data.frame(CPM_table)
plotMDS(y)
log_y<-rlog(y)

class(y)
View(spacefly.counts)
###### the above is just the codes for generating the transcript-based counts ######
###### From here, I start to do the gene-based counts ######


source("https://bioconductor.org/biocLite.R")
biocLite("tximport")
biocLite("biomaRt")
library(biomaRt)  
library(tximport)  

setwd("C:/Users/JiyingLi/Desktop/project221")
flymatrix<-read.csv("flies_space_counts.csv")
rownames(flymatrix)<-flymatrix$Name
flymatrix<-flymatrix[,c(2:9)]
ensembl=useMart("ensembl")
ensembl_id<-rownames(flymatrix)
ensembl = useDataset("dmelanogaster_gene_ensembl",mart=ensembl)
query<-getBM(attributes=c('flybase_transcript_id','ensembl_gene_id','external_gene_name','description'), filters = 'flybase_transcript_id', values = ensembl_id, mart=ensembl) # some of the parameters here I am not aware of. Are they related the fly genome info or something?
View(query)  # query is table generated by transcript to gene, 4column： txid,geneid,genename,geneannotation
tx2gene<-query[,c(1,2)] # only keep the 1st and 2nd column: txid to geneid correspondence
colnames(tx2gene)<-c("TXNAME","GENEID")
dir<-"C:/Users/JiyingLi/ECE221_final_project" # convenient, assign the address to a variable
list.files("C:/Users/JiyingLi/ECE221_final_project")  # list.files = ls, listing all the files in the folder
files <- file.path(dir, "salmon",c("SRR3390478.quant","SRR3390479.quant","SRR3390480.quant","SRR3390481.quant","SRR3390482.quant","SRR3390483.quant","SRR3390484.quant","SRR3390485.quant"), "quant.sf")
files # files，shows all the individual file, created by salmon
names(files) <- c("G3R3","G1R2","G1R3","G1R4","G3R1","G3R2","G1R1","G3R4") #tag each file
files
install.packages(readr)  // need to have the double quote for the pkg name
install.packages("readr")
library(readr) 
txi <- tximport(files, type = "salmon", tx2gene = tx2gene, reader = read_tsv)
head(txi$counts)
dim(txi)
dim(txi$counts)
write.csv(txi$counts, "space_fly_gene_counts.csv")
savehistory("C:/Users/JiyingLi/Desktop/project221/11.22.16")
### until here, I am done with processing the transcript-based counts to gene-based counts.
### I can continue to analyze these data. Or I can stop here, and resume starting with the space_fly_gene_counts.csv, since all the data needed for the EdgeR analysis.


###Lisa's recommend###
#Then, for edgeR

library(edgeR)
cts <- txi$counts     # txi contains counts data
normMat <- txi$length   # txi also contains length data, 去点开salmon文件夹里的data就可以看到原始数据。
normMat <- normMat/exp(rowMeans(log(normMat))) # normalization step
library(limma)
library(edgeR)
o <- log(calcNormFactors(cts/normMat)) + log(colSums(cts/normMat))
y <- DGEList(cts)
y$offset <- t(t(log(normMat)) + o)
# then back to the EdgeR methods.

Groups <- c("G3", "G1", "G1", "G1", "G3", "G3", "G1", "G3")
Groups <- c("G3", "G1", "G1", "G1", "G3", "G3", "G1", "G3")
y <- DGEList(counts=cts，group=factor(Groups) )
y$offset <- t(t(log(normMat)) + o)

bcv <- 0.4
CPM <- cpm(y)
CPM_table<-cbind(CPM,rownames(CPM))
CPM_table<-as.data.frame(CPM_table)
# colnames(CPM_table)<-c("sal25ppt","sal30ppt","sal35ppt","contig_ID")
plotMDS(y, labels=colnames(y), cex = .8, col=as.numeric(y$samples$group)) 

### in Stanford course, there are more parameters, such as method="bcv" ###
plotMDS(d, method="bcv", col=as.numeric(d$samples$group))
legend("bottomleft", as.character(unique(d$samples$group)), col=1:3, pch=20)

###Before doing DEG analysis, we can do some dispersion analysis:
###one way: we use empirical Bayes tagwise dispersions
y1 <- estimateCommonDisp(y, verbose=T)
names(y1)
y1 <- estimateTagwiseDisp(y1)
names(y1)
plotBCV(y1)

###another way: GLM estimates of dispersion (in EdgeR)
design.mat <- model.matrix(~ 0 + d$samples$group)
colnames(design.mat) <- levels(y$samples$group)
y2 <- estimateGLMCommonDisp(y,design.mat)
y2 <- estimateGLMTrendedDisp(y2,design.mat, method="power")
# You can change method to "auto", "bin.spline", "power", "spline", "bin.loess".
# The default is "auto" which chooses "bin.spline" when > 200 tags and "power" otherwise.
y2 <- estimateGLMTagwiseDisp(y2,design.mat)
plotBCV(y2)

### Until here, the PCA figure is finished ###
### Need to generate the volcano figure

## start the DEG
G1G3 <- exactTest(y1, pair=c(1,2))
topTags(G1G3, n=10)
G1G3filter<-decideTestsDGE(G1G3, adjust.method="BH", p.value=0.05)
summary(G1G3filter) # get the DEG data
G1G3tag <- rownames(y1)[as.logical(G1G3filter)] 
plotSmear(G1G3, de.tags=G1G3tag, main="Space fly hypergravity RNAseq (EdgeR)")
abline(h = c(-2, 2), col = "blue")
#DONE

### get the DEG result data ###
head(G1G3filter)
z <- DGEList(txi$counts)
z <- calcNormFactors(z)
design <- model.matrix(~condition, data = sampleTable)
z1 <- estimateCommonDisp(z, verbose=T)
names(z1)
z1 <- estimateTagwiseDisp(z1)
plotBCV(z1)
G1G3_limma <- exactTest(z1, pair=c(1,2))
head(G1G3filter)
head(G1G3$table)
head(G1G3$genes)
head(G1G3$comparison)
head(G1G3$table)
write.csv(G1G3$table, "result1.csv")



###### have not done limma #######
##And for limma have not done yet
library(limma)
y <- DGEList(txi$counts)
y <- calcNormFactors(y)
design <- model.matrix(~condition, data = sampleTable)
v <- voom(y, design)




